File: Mathematics/palindrome/JavaScript/palindrome.js
function palindrome(str) {
    str = str.toLowerCase();
    str = str.replace(/[\s_,.()|:-]/g, "");
    
    let newStr = str.split("").reverse().join("");
    if(newStr === str){
        output = true;
    }
    else{
        output = false;
    }
    console.log(output);
}

palindrome("eye");
palindrome("_eye");
palindrome("race car");
palindrome("not a palindrome");
palindrome("A man, a plan, a canal. Panama");
palindrome("never odd or even");
palindrome("nope");
palindrome("almostomla");
palindrome("My age is 0, 0 si ega ym.");
palindrome("1 eye for of 1 eye.")
palindrome("0_0 (: /-\ :) 0-0");
palindrome("five|\_/|four");

File: Mathematics/palindrome/ruby/palindrome.rb
# Validates if a string with all punctuation removed
# reads the same backwards as forwards
def palindrome?(str)
  # Convert string to lower case and emove all non-word characters
  sanitized_str = str.downcase.gsub(/\W/, '')
  sanitized_str == sanitized_str.reverse
end

strs = [
  'eye',
  'race car',
  'not a palindrome',
  'A man, a plan, a canal. Panama',
  'never odd or even',
  'nope',
  'My age is 0, 0 si ega ym.',
]
strs.each do |str|
  puts "#{palindrome?(str)}\t| #{str}"
end
